service/src/NVMLDevicePool.cpp

```diff
@@ -150,6 +150,30 @@ namespace geopm
         return (uint64_t)result;
     }
 
+    std::vector<unsigned int> NVMLDevicePoolImp::frequency_supported_sm(int accel_idx) const
+    {
+        check_accel_range(accel_idx);
+        unsigned int count = 100;
```

Comment:
Where is this magic number coming from? Prefer if it was a static const member.

Reply:
It's just that, a magic number to start with that is a reasonable size, similar to M_MAX_CONTEXTS.  

We can query the return size by passing count as 0, but this will result in always calling the NVML library twice, which can be time consuming (though we have not check the time of this particular function call).